File: ios/DriversTest/DriversTest/view/screen/ResultsScreen.swift
import SwiftUI

struct ResultsScreen: View {
    @ObservedObject var vm: QuizViewModel
    @ObservedObject var localizer: Localizer

    var body: some View {
        ScrollView {
            VStack(spacing: 16) {
                // Score circle
                ZStack {
                    Circle()
                        .stroke(vm.didPass ? AppTheme.green : AppTheme.red, lineWidth: 6)
                        .frame(width: 160, height: 160)

                    VStack(spacing: 2) {
                        Text("\(vm.resultPct)%")
                            .font(.system(size: 42, weight: .bold))
                            .monospacedDigit()
                            .foregroundColor(vm.didPass ? AppTheme.green : AppTheme.red)
                        Text(vm.didPass ? localizer.localized("pass") : localizer.localized("fail"))
                            .font(.system(size: 14, weight: .semibold))
                            .foregroundColor(vm.didPass ? AppTheme.green : AppTheme.red)
                            .textCase(.uppercase)
                    }
                }
                .padding(.top, 24)

                Text(vm.didPass ? localizer.localized("congratulations") : localizer.localized("keepPracticing"))
                    .font(.system(size: 22, weight: .bold))

                if let state = vm.currentState {
                    Text(localizer.localized("resultDetail", substitutions: [
                        "correct": "\(vm.correctCount)",
                        "total": "\(vm.questions.count)",
                        "pass_pct": "\(state.passingScorePct)",
                        "agency": state.agency,
                    ]))
                    .font(.system(size: 15))
                    .foregroundColor(AppTheme.gray)
                    .multilineTextAlignment(.center)
                    .lineSpacing(3)
                }

                // Action buttons
                Button {
                    vm.goHome()
                } label: {
                    Text(localizer.localized("newQuiz"))
                        .font(.system(size: 18, weight: .semibold))
                        .foregroundColor(AppTheme.onPrimary)
                        .frame(maxWidth: .infinity)
                        .padding(16)
                        .background(AppTheme.blue)
                        .clipShape(RoundedRectangle(cornerRadius: 12))
                }

                Button {
                    vm.showStats()
                } label: {
                    Text(localizer.localized("viewStats"))
                        .font(.system(size: 18, weight: .semibold))
                        .foregroundColor(AppTheme.blue)
                        .frame(maxWidth: .infinity)
                        .padding(16)
                        .cardStyle(borderColor: AppTheme.blue)
                }

                // Review section
                if !vm.wrongResults.isEmpty {
                    VStack(alignment: .leading, spacing: 12) {
                        Text(localizer.localized("reviewMissed", substitutions: ["count": "\(vm.wrongResults.count)"]))
                            .font(.system(size: 18, weight: .semibold))

                        ForEach(vm.wrongResults, id: \.id) { result in
                            ReviewItemView(result: result, localizer: localizer)
                        }
                    }
                    .frame(maxWidth: .infinity, alignment: .leading)
                    .padding(.top, 8)
                } else {
                    VStack(spacing: 8) {
                        Text(localizer.localized("perfectScore"))
                            .font(.system(size: 18, weight: .semibold))
                        Text(localizer.localized("perfectMsg"))
                            .font(.system(size: 15))
                            .foregroundColor(AppTheme.gray)
                    }
                    .padding(.top, 8)
                }
            }
            .padding(16)
        }
        .background(AppTheme.grayLight)
    }
}

File: android/app/src/main/java/com/drivers/test/view/screen/ResultsScreen.kt
package com.drivers.test.view.screen

import androidx.compose.foundation.background
import androidx.compose.foundation.border
import androidx.compose.foundation.layout.*
import androidx.compose.foundation.rememberScrollState
import androidx.compose.foundation.shape.CircleShape
import androidx.compose.foundation.shape.RoundedCornerShape
import androidx.compose.foundation.verticalScroll
import androidx.compose.material3.MaterialTheme
import androidx.compose.material3.Text
import androidx.compose.runtime.Composable
import androidx.compose.ui.Alignment
import androidx.compose.ui.Modifier
import androidx.compose.ui.draw.clip
import androidx.compose.ui.text.font.FontStyle
import androidx.compose.ui.text.font.FontWeight
import androidx.compose.ui.text.style.TextAlign
import androidx.compose.ui.unit.dp
import androidx.compose.ui.unit.sp
import com.drivers.test.model.SessionResult
import com.drivers.test.theme.AppTheme
import com.drivers.test.view.components.PrimaryButton
import com.drivers.test.view.components.SecondaryButton
import com.drivers.test.viewmodel.QuizViewModel

@Composable
fun ResultsScreen(vm: QuizViewModel) {
    val c = AppTheme.colors
    val state = vm.currentState ?: return

    Column(
        modifier = Modifier
            .fillMaxSize()
            .background(MaterialTheme.colorScheme.background)
            .verticalScroll(rememberScrollState())
            .padding(16.dp),
        horizontalAlignment = Alignment.CenterHorizontally,
    ) {
        Spacer(Modifier.height(24.dp))

        // Score circle
        Box(
            modifier = Modifier
                .size(160.dp)
                .border(6.dp, if (vm.didPass) c.green else c.red, CircleShape),
            contentAlignment = Alignment.Center,
        ) {
            Column(horizontalAlignment = Alignment.CenterHorizontally) {
                Text(
                    "${vm.resultPct}%",
                    fontSize = 42.sp,
                    fontWeight = FontWeight.Bold,
                    color = if (vm.didPass) c.green else c.red,
                )
                Text(
                    if (vm.didPass) vm.t("pass") else vm.t("fail"),
                    fontSize = 14.sp,
                    fontWeight = FontWeight.SemiBold,
                    color = if (vm.didPass) c.green else c.red,
                )
            }
        }

        Spacer(Modifier.height(24.dp))

        Text(
            if (vm.didPass) vm.t("congratulations") else vm.t("keepPracticing"),
            fontSize = 22.sp,
            fontWeight = FontWeight.Bold,
        )
        Spacer(Modifier.height(8.dp))
        Text(
            vm.t(
                "resultDetail",
                mapOf(
                    "correct" to "${vm.correctCount}",
                    "total" to "${vm.questions.size}",
                    "pass_pct" to "${state.passingScorePct}",
                    "agency" to state.agency,
                ),
            ),
            fontSize = 15.sp,
            color = c.gray,
            textAlign = TextAlign.Center,
            lineHeight = 22.sp,
        )

        Spacer(Modifier.height(24.dp))

        PrimaryButton(vm.t("newQuiz")) { vm.goHome() }
        Spacer(Modifier.height(10.dp))
        SecondaryButton(vm.t("viewStats")) { vm.showStats() }

        Spacer(Modifier.height(24.dp))

        // Review section
        if (vm.wrongResults.isNotEmpty()) {
            Text(
                vm.t("reviewMissed", mapOf("count" to "${vm.wrongResults.size}")),
                fontSize = 18.sp,
                fontWeight = FontWeight.SemiBold,
                modifier = Modifier.fillMaxWidth(),
            )
            Spacer(Modifier.height(12.dp))
            vm.wrongResults.forEach { result ->
                ReviewItem(result, vm)
                Spacer(Modifier.height(10.dp))
            }
        } else {
            Text(vm.t("perfectScore"), fontSize = 18.sp, fontWeight = FontWeight.SemiBold)
            Spacer(Modifier.height(8.dp))
            Text(vm.t("perfectMsg"), fontSize = 15.sp, color = c.gray)
        }
    }
}

@Composable
private fun ReviewItem(
    result: SessionResult,
    vm: QuizViewModel,
) {
    val c = AppTheme.colors
    Row(
        modifier = Modifier
            .fillMaxWidth()
            .clip(RoundedCornerShape(12.dp))
            .background(c.card),
    ) {
        Box(modifier = Modifier.width(4.dp).fillMaxHeight().background(c.red))
        Column(modifier = Modifier.padding(14.dp)) {
            Text(result.question, fontSize = 14.sp, fontWeight = FontWeight.SemiBold)
            Spacer(Modifier.height(6.dp))
            Text(
                "${vm.t("yourAnswer")}: ${result.yourAnswer}: ${result.yourAnswerText}",
                fontSize = 13.sp,
                color = c.gray,
            )
            Text(
                "${vm.t("correct")}: ${result.correctAnswer}: ${result.correctAnswerText}",
                fontSize = 13.sp,
                color = c.gray,
            )
            Spacer(Modifier.height(6.dp))
            Text(result.explanation, fontSize = 13.sp, color = c.gray, fontStyle = FontStyle.Italic)
        }
    }
}
